Examples/Examples/All Examples/SceneKitExample.swift

```diff
@@ -0,0 +1,227 @@
+            mapOptions.render.presentsWithTransaction = true
+        }
+
+        self.mapView.on(.styleLoadingFinished) { [weak self] _ in
+            self?.addModelAndTerrain()
+        }
+    }
+
+    override public func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+         // The below line is used for internal testing purposes only.
+        self.finish()
+    }
+    
+    func addModelAndTerrain() {
+        try! mapView.__map.addStyleCustomLayer(forLayerId: "Custom",
+                                                layerHost: self,
+                                            layerPosition: nil)
+                                            
+        var demSource = RasterDemSource()
+        demSource.url = "mapbox://mapbox.mapbox-terrain-dem-v1"
+        demSource.tileSize = 512
+        demSource.maxzoom = 14.0
+        mapView.style.addSource(source: demSource, identifier: "mapbox-dem")
+        let terrain = Terrain(sourceId: "mapbox-dem");
+        _ = self.mapView.style.setTerrain(terrain)
+
+        var skyLayer = SkyLayer(id: "sky-layer")
+        skyLayer.paint?.skyType = .atmosphere
+        skyLayer.paint?.skyAtmosphereSun = .constant([0, 0])
+        skyLayer.paint?.skyAtmosphereSunIntensity = .constant(15.0)
+
+        _ = self.mapView.style.addLayer(layer: skyLayer)
+        
+        // Re-use terrain source for hillshade
+        let map = self.mapView.__map!
+        let properties = [
+            "id": "terrain_hillshade",
+            "type": "hillshade",
+            "source": "mapbox-dem",
+            "hillshade-illumination-anchor": "map"
+        ] as [String : Any]
+            
+        let insertHillshadeBelow = try! map.styleLayerExists(forLayerId: "water") ?
+            LayerPosition(above: nil, below: "water", at: nil) : try! map.styleLayerExists(forLayerId: "hillshade") ?
+            LayerPosition(above: nil, below: "hillshade", at: nil) : nil;
+        try! map.addStyleLayer(forProperties: properties, layerPosition: insertHillshadeBelow)
+    }
+
+    public func renderingWillStart(_ metalDevice: MTLDevice, colorPixelFormat: UInt, depthStencilPixelFormat: UInt) {
+        renderer = SCNRenderer(device:metalDevice)
+        scene = SCNScene()
+        renderer.scene = scene
+
+        modelNode = SCNScene(named:"34M_17")?.rootNode;
+        scene.rootNode.addChildNode(modelNode)
+        
+        cameraNode = SCNNode();
+        let camera = SCNCamera();
+        cameraNode.camera = camera;
+        camera.usesOrthographicProjection = false;
+        scene.rootNode.addChildNode(cameraNode);
+        renderer.pointOfView = cameraNode;
+        self.setupLight()
+        // In order to use depth occlusion, align with gl-native Z handling (doesn't use reverse Z).
+        if #available(iOS 13.0, *) {
+            renderer.usesReverseZ = false
+        } else {
+            // Fallback on earlier versions, disable depth in render()
```

Comment:
What happens in these cases? Should we disable this example completely if < iOS 13?

Reply:
We should enable it, just there is a need to use CPU side occlusion (the same approach that could be used when geo-referencing UI elements to map). The idea is to disable depth buffer (now disabled in demo) and show or hide model with animated fade in / fade out (similar to behavior of markers in gl-js).

![90252440-2cde5000-de48-11ea-8f3f-6bffa52e07e1](https://user-images.githubusercontent.com/549216/109847300-b2e0ad80-7c57-11eb-8605-8da44f2832ee.gif)

As we cannot have occlusion done yet, current behavior on iOS < 13 is like here (model always visible):

https://user-images.githubusercontent.com/549216/109847499-ee7b7780-7c57-11eb-8e3d-e33b9c1c0f75.mov